src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -403,16 +419,17 @@ impl<T: 'static> EventLoop<T> {
             // woken up by messages arriving from the Wayland socket, to avoid delaying the
             // dispatch of these events until we're woken up again.
             let instant_wakeup = {
-                let handle = self.event_loop.handle();
-                let source = self.wayland_source.clone();
-                let dispatched = handle.with_source(&source, |wayland_source| {
-                    let queue = wayland_source.queue();
-                    self.with_state(|state| {
-                        queue.dispatch_pending(state, |_, _, _| unimplemented!())
-                    })
-                });
+                let mut wayland_source = self.wayland_dispatcher.as_source_mut();
+                let queue = wayland_source.queue();
+                let state = match &mut self.window_target.p {
+                    crate::platform_impl::EventLoopWindowTarget::Wayland(ref mut window_target) => {
```

Comment:
Maybe import part of this? A fully qualified path like this seems somewhat unusual. And is the `ref mut` necessary? Would have expected this to work just because of the `&mut self.window_target.p`.